bindings/python/pymongoarrow/types.py

```diff
@@ -30,20 +31,35 @@ class _BsonArrowTypes(enum.Enum):
     string = 6
 
 
+# Custom Extension Types.
+
+class ObjectIdType(PyExtensionType):
+    _type_marker = _BsonArrowTypes.objectid
+
+    def __init__(self):
+        PyExtensionType.__init__(self, binary(12))
+
+    def __reduce__(self):
+        return ObjectIdType, ()
```

Comment:
Any particular reason you chose to use reduce here https://docs.python.org/3/library/pickle.html#object.__setstate__

> Although powerful, implementing __reduce__() directly in your classes is error prone. For this reason, class designers should use the high-level interface (i.e., __getnewargs_ex__(), __getstate__() and __setstate__()) whenever possible. We will show, however, cases where using __reduce__() is the only option or leads to more efficient pickling or both.

Also should we test that this actually works?

Reply:
I pulled this pattern from the arrow [docs](https://arrow.apache.org/docs/python/extending_types.html#defining-extension-types-user-defined-types).  I'll use the advanced `__arrow_ext_serialize__` method instead.    Agreed we should test it either way.

Reply:
Oh in that case can you just add a comment referencing that docs page? I don't think we need to use the longer `__arrow_ext_serialize__` methods yet since we don't need their functionality.

Reply:
Done and tested